### DISCUSSION TITLE
Out of memory error when using with KiCAD?

### DISCUSSION THREAD

**Discussion by jacks178** [freerouting/freerouting#159] [General]
Hi, I'm trying all sorts of things to make Freerouting work with KiCAD (I'm currently using a 6.99 from a few days ago fwiw). I am running Windows 10. Whenever I try to import a dsn file to Freerouting either as a plugin in KiCAD or in its own window, it thinks for a while before alluding to Java being out of memory with the error "java.lang.OutOfMemoryError: Java heap space: failed reallocation of scalar replaced objects". It doesn't seem to matter what size the file is, and the one I've included below is giving the error despite not being very complicated.

Does anyone know what else I can do to get this program working? Thanks!

[setonix_2hp-pcb.zip](https://github.com/freerouting/freerouting/files/10329558/setonix_2hp-pcb.zip)

**Comment by andrasfuchs** [comment]
It seems to be working for me with version 1.6.5 on Windows 11.

What version of Java JRE do you have? Could you include the logs?

**Comment by jacks178** [comment]
I'm running (or trying to run) 1.6.5 on Windows 10. I have the latest download of Java JRE (Version 8 Update 351). I've never used JRE before so I don't know how to include the logs since KiCAD 6 is telling me to close JRE to run the program. Once I close JRE, the error looks like this:

Program failure:
---
command:
java -jar
(Jar destination)
---exit code:3221225786
--- stdout ---
b''
--- stderr ---
b''

**Reply by andrasfuchs** [reply]
I think the problem might be that you have Java 8 installed, and not the required Java 17. Could you install it [from here](https://adoptium.net/temurin/releases) and try again?

**Comment by jacks178** [comment]
Ah, that appears to have done the trick and the program is running now. Thank you!
